go/runtimestats/stats.go

```diff
@@ -0,0 +1,137 @@
+package runtimestats
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+	"sync"
+	"time"
+
+	"github.com/keybase/client/go/chat/globals"
+	"github.com/keybase/client/go/chat/utils"
+	"github.com/keybase/client/go/protocol/keybase1"
+	"golang.org/x/sync/errgroup"
+)
+
+type Runner struct {
+	globals.Contextified
+	sync.Mutex
+
+	started bool
+	stopCh  chan struct{}
+	eg      errgroup.Group
+}
+
+func NewRunner(g *globals.Context) *Runner {
+	return &Runner{
+		Contextified: globals.NewContextified(g),
+	}
+}
+
+func (r *Runner) debug(ctx context.Context, msg string, args ...interface{}) {
+	r.G().Log.CDebugf(ctx, "RuntimeStats.Runner: %s", fmt.Sprintf(msg, args...))
+}
+
+func (r *Runner) Start(ctx context.Context) {
+	defer r.G().CTrace(ctx, "Runner.Start", func() error { return nil })()
+	r.Lock()
+	defer r.Unlock()
+	if !r.G().Env.GetRuntimeStatsEnabled() {
+		r.debug(ctx, "not starting, not enabled in env")
+		return
+	}
+	if r.started {
+		return
+	}
+	r.stopCh = make(chan struct{})
+	r.started = true
+	r.eg.Go(func() error { return r.statsLoop(r.stopCh) })
+	r.G().PushShutdownHook(func() error {
```

Comment:
should do this in the `NewRunner` function or it can get pushed on multiple times